test/alternator/test_batch.py

```diff
@@ -311,6 +311,21 @@ def test_batch_get_item_projection_expression(test_table):
         expected_items = [{k: item[k] for k in wanted if k in item} for item in items]
         assert multiset(got_items) == multiset(expected_items)
 
+# ExpressionAttributeNames can be used to define aliases for attribute names for ProjectionExpression
+def test_batch_get_item_projection_expression_with_substitute_name(test_table):
+    items = [{'p': random_string(), 'c': random_string(), 'val2': random_string()} for i in range(10)]
+    with test_table.batch_writer() as batch:
+        for item in items:
+            batch.put_item(item)
+    keys = [{k: x[k] for k in ('p', 'c')} for x in items]
+    for wanted in [['p'], ['p', 'c'], ['p', 'val2']]:
+        wantedStr = ",".join(wanted).replace('p', '#ppp')
```

Comment:
I think this test is needlessly complicated because you copy-pasted the previous one whose intention was to check multiple attributes. I think here you can just read one attribute instead of doing this general-looking code.

Also, you forgot to add a check for the error cases: missing ExpressionAttributeNames (e.g., you ask for "#xyz" but it's not defined) and spurious ExpressionAttributeNames (e.g., ExpressionAttributeNames defines "#abc" but it's not used in the projection expression).

Reply:
Oh, I see you added a separate test for the spurious expressions in the next patch.

It doesn't have to be a separate test (you can do multiple checks in the same test, if you don't overdo it), but your choice.

Reply:
merged